apps/arena/native/physics/src/collision_detection/ear_clipping.rs

```diff
@@ -0,0 +1,197 @@
+            get_previous_cyclic_index(current_vertex_index, &polygon.vertices);
+        let next_vertex_index = get_next_cyclic_index(current_vertex_index, &polygon.vertices);
+
+        let previous_vertex = polygon.vertices[previous_vertex_index];
+        let current_vertex = polygon.vertices[current_vertex_index];
+        let next_vertex = polygon.vertices[next_vertex_index];
+
+        if !is_vertex_consecutive(&previous_vertex, &current_vertex, &next_vertex) {
+            cleaned_vertices.push(current_vertex);
+        }
+    }
+
+    polygon.vertices = cleaned_vertices;
+}
+
+fn is_vertex_consecutive(
+    previous_vertex: &Position,
+    current_vertex: &Position,
+    next_vertex: &Position,
+) -> bool {
+    let first_vector = Position::sub(previous_vertex, current_vertex);
+    let second_vector = Position::sub(next_vertex, current_vertex);
+
+    get_cross_product_scalar(&first_vector, &second_vector) == 0.0
+}
+
+fn triangulate_polygon(polygon: &Entity) -> Vec<Entity> {
+    let mut result: Vec<Entity> = vec![];
+    let mut positions = polygon.vertices.clone();
+    let mut ear_found: bool;
+
+    while positions.len() > 3 {
+        ear_found = false;
+        for current_vertex_index in 0..positions.len() {
+            let previous_vertex_index = get_previous_cyclic_index(current_vertex_index, &positions);
+            let next_vertex_index = get_next_cyclic_index(current_vertex_index, &positions);
+
+            let previous_vertex = positions[previous_vertex_index];
+            let current_vertex = positions[current_vertex_index];
+            let next_vertex = positions[next_vertex_index];
+
+            if is_position_ear(current_vertex, previous_vertex, next_vertex, &positions) {
+                ear_found = true;
+                let triangle_vertex = vec![current_vertex, next_vertex, previous_vertex];
+                let candidate_triangle = Entity::new_polygon(1, triangle_vertex);
+                result.push(candidate_triangle);
+                positions.retain(|pos| pos != &current_vertex);
+                break;
+            }
+        }
+
+        if !ear_found {
+            panic!("No ear detected, polygon invalid")
+        }
+    }
+    let previous_vertex = positions[0];
+    let current_vertex = positions[1];
+    let next_vertex = positions[2];
+
+    let triangle_vertex = vec![current_vertex, next_vertex, previous_vertex];
+    let candidate_triangle = Entity::new_polygon(1, triangle_vertex);
+    result.push(candidate_triangle);
+
+    // This algorithm should always result n - 2 triangles where n is the amount of vertex
+    // in the polygon
+    if result.len() > polygon.vertices.len() - 2 {
+        panic!("Wrong triangulation result")
+    }
+
+    result
+}
+
+// Check if the three vertex are a valid ear, this means
+// 1. The inner angle formed by the three vertex isn't greater that 180 degrees
+// 2. There isn't any vertex inside the triangle area
+fn is_position_ear(
```

Comment:
is position ear? I think an ear is a triangle (3 vertices). Please take a look at the names because they're not following the Math concepts.

Reply:
Yeah since we're using positions here, i got confused. I'll build the triangle before doing thing and use that instead of a position, so the name `is_triangle_ear` has sense